plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeTableChangesWithAccessTracking.java

```diff
@@ -0,0 +1,114 @@
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(TEST_SESSION).build();
+
+        dataDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("delta_lake").toFile();
+        metastore = createTestingFileHiveMetastore(dataDirectory);
+        fileSystemFactory = new AccessTrackingFileSystemFactory(new HdfsFileSystemFactory(HDFS_ENVIRONMENT));
+
+        queryRunner.installPlugin(new TestingDeltaLakePlugin(Optional.of(new TestingDeltaLakeMetastoreModule(metastore)), Optional.of(fileSystemFactory), EMPTY_MODULE));
+        queryRunner.createCatalog(
+                "delta_lake",
+                "delta_lake",
+                ImmutableMap.of("delta.enable-non-concurrent-writes", "true"));
+
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA);
+        return queryRunner;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void cleanup()
+            throws IOException
+    {
+        if (metastore != null) {
+            metastore.dropDatabase(SCHEMA, false);
+            deleteRecursively(dataDirectory.toPath(), ALLOW_INSECURE);
+        }
+    }
+
+    @Test
+    public void testTableChangesFileSystemAccess()
+    {
+        String tableName = "test_reading_ranges_of_changes_on_table_with_cdf_enabled_" + randomNameSuffix();
+        try {
+            assertUpdate("CREATE TABLE " + tableName + " (page_url VARCHAR, domain VARCHAR, views INTEGER) WITH (change_data_feed_enabled = true)");
+            assertUpdate("INSERT INTO " + tableName + " VALUES('url1', 'domain1', 1)", 1);
+            assertUpdate("INSERT INTO " + tableName + " VALUES('url2', 'domain2', 2)", 1);
+            assertUpdate("INSERT INTO " + tableName + " VALUES('url3', 'domain3', 3)", 1);
+            assertUpdate("UPDATE " + tableName + " SET page_url = 'url22' WHERE domain = 'domain2'", 1);
+            assertUpdate("UPDATE " + tableName + " SET page_url = 'url33' WHERE views = 3", 1);
+            assertUpdate("DELETE FROM " + tableName + " WHERE page_url = 'url1'", 1);
+            assertQuery("SELECT * FROM " + tableName, "VALUES('url22', 'domain2', 2), ('url33', 'domain3', 3)");
+
+            fileSystemFactory.resetOpenCount();
+            assertQuerySucceeds("SELECT * FROM TABLE(system.table_changes('" + SCHEMA + "', '" + tableName + "', 0))");
+            assertThat(fileSystemFactory.getOpenCount()).containsAllEntriesOf(
+                    ImmutableMap.of(
+                            "00000000000000000001.json", 1,
+                            "00000000000000000002.json", 1,
+                            "00000000000000000003.json", 1,
+                            "00000000000000000004.json", 1,
+                            "00000000000000000005.json", 1,
+                            "00000000000000000006.json", 1,
+                            "00000000000000000007.json", 2));
+        }
+        finally {
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
```

Comment:
I'd omit IF EXISTS and start the `try` block only after CREATE TABLE succeeds.
in fact, I wouldn't use try finally at all, because the cleanup isn't very important here (local isolated test)